pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -309,12 +311,20 @@ func (prs *PipelineRunStatus) ConvertFrom(ctx context.Context, source *v1.Pipeli
 		new.convertFrom(ctx, st)
 		prs.SkippedTasks = append(prs.SkippedTasks, new)
 	}
-	prs.ChildReferences = nil
-	for _, cr := range source.ChildReferences {
-		new := ChildStatusReference{}
-		new.convertFrom(ctx, cr)
-		prs.ChildReferences = append(prs.ChildReferences, new)
+	embeddedStatus := config.FromContextOrDefaults(ctx).FeatureFlags.EmbeddedStatus
+	if embeddedStatus == config.BothEmbeddedStatus || embeddedStatus == config.MinimalEmbeddedStatus {
+		prs.ChildReferences = nil
+		for _, cr := range source.ChildReferences {
+			new := ChildStatusReference{}
+			new.convertFrom(ctx, cr)
+			prs.ChildReferences = append(prs.ChildReferences, new)
+		}
+	}
+	if embeddedStatus == config.BothEmbeddedStatus || embeddedStatus == config.FullEmbeddedStatus {
+		prs.TaskRuns = taskRuns
```

Comment:
nit: we might not need `cmpopts.EquateEmpty()` added in all test casese if we check the length of serialized taskRuns and runs?
```
		if len(taskRuns) != 0 {
			prs.TaskRuns = taskRuns
		}
		if len(runs) != 0 {
			prs.Runs = runs
		}
```

Reply:
Here's the closest thing I found in the go style guide: https://google.github.io/styleguide/go/decisions#nil-slices
Doesn't seem like there's clear guidance in this situation so I'm going to go with my personal preference, to keep EquateEmpty. (I think our tests shouldn't care whether there's an empty slice or nil pointer.)

Reply:
Thanks Lee! Makes perfect sense.